### ISSUE TITLE
Archi Database Plugin can not initialize

### ISSUE THREAD

**Issue by gitPrasun** [archi-contribs/database-plugin#111]
The Archi 4.7.1 needed new version of mysql 8- hence installed and configured new , however while initializing the database it fails due to the attached error:

it creates 3 empty tables : database_version, folders,models before throwing the attached error.
the database version: mysql-8.0.22-winx64

is this usual? any known remedies please.
![archi error](https://user-images.githubusercontent.com/33220251/99314994-3ec5f980-2862-11eb-8aeb-de98d1ea5aa9.PNG)

**Comment by herve91** [comment]
Hi,

I'll have a look and send a fix asap.

Best regards
Hervé

**Comment by herve91** [comment]
Hi,

MySQL 8.0 introduces a new function called rank(). Unfortunately, I've got a column called rank in my tables which is therefore not allowed anymore in version 8.

Why do you state Archi 4.7.1 needs new version of MySQL 8 ? Can you use MySQL 7 or MariaDB to avoid this new function rank () ?

To be honest, I do not have time at the moment to change my database model ...

**Comment by gitPrasun** [comment]
Hi,
My entire post was due to the fact Archi 4.7.1 could not work with  mysql-5.7.17-winx64. ( which was working fine with version 4.4)
During the "Export Model to database" it throws error
![image](https://user-images.githubusercontent.com/33220251/99950513-e9677c00-2d7c-11eb-8813-f9352f31e929.png)

I have no option remaining - as Archi 4.7.1 does not let me use it . also as I updated the model I can not use any lower version of Archi - Practically I am stuck :(.

**Comment by gitPrasun** [comment]
**Updates :** I am able to initialize  using version mysql-5.7.32-winx64 which is later than my current 5.7 but still not 8 with Archi 4.7.1

however I still get an error 
![image](https://user-images.githubusercontent.com/33220251/99971811-ff396900-2d9d-11eb-9d4f-a756e6d5a00a.png)

I understand this is importing in relationships and stuck. I am open for manual way as well - can you please suggest how can I get 'Strength' using ALASQL ? my target would be to remove the strengths for which it fails.

Thanks in advance.

**Comment by gitPrasun** [comment]
**Update 2:** Removed 'Strength' from the .archimate file - and it worked ( piece of cake) 🥇 . :) 
I can close the issue now as I could still live till you can make the support mySQL9 - once you have time . or do you want to have it open for tracking ?

**Comment by herve91** [comment]
Hi,
I'm doing my tests with MariaDB 15.1 therefore my plugin should definitively work with MySQL 5.7.x. If not, this means that I missed a difference between MariaDB 15.1 and MySQL 5.7.

To help me narrowing my tests, could your please confirm if the error message **with MySQL 5.7** was when you tried to upgrade an existing database the new plugin version or when you create a new database ?

**Comment by herve91** [comment]
I re-read your post and I believe I understand what happens ... migrating to MySQL 8 won't solve your issue.

The Archimate definition of strength is:
`Attributes can be used to indicate the sign and/or strength of the influence. The choice of possible attribute values is left to the modeler; e.g., {++, +, 0, -, --} or [0..10].`

I know that Archimate does not enforce short expressions for strengthes but the examples given are so small that I implemented the strength column as a varchar(20) which is obviously too short for you.

I will update the datamodel to allow longer strengthes and this will fix your issue.

**Comment by gitPrasun** [comment]
Well it certainly worked for mysql-5.7.32-winx64 but not for mysql-5.7.17-winx64 - both are 5.7 , I guess.

with mysql-5.7.17-winx64( which i already setup with Archiv4.4) I got error ( while using Archi4.7.1)when 
1) I tried to use it ( it did not let me pass without an upgrade) - & subsequently 
2) I tried to upgrade the existing database. ( I got the 2nd error)

I thought mySQL8 would be the choice and i set it up fresh , but
3) I got the initialize error. 
at this point I was totally stuck and posted the bug to you.( I hope I posted all the screenshots)

I kept on trying with the latest minor release for 5.7 which is mysql-5.7.32-winx64 and it worked for now.
